### PR TITLE
Refactor submit buttons common alerts#12740

### PR DESCRIPTION
**Pull request by Clark-W-Griswold**
## Description
Issue: #[12740](https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/12740)
In the `SubmitButtons` component, we have repeated code that functions as the visual representation of an error/alert. The purpose of this PR is to extract that into it's own reusable component and then update `SubmitButtons` to use this new component. 

## Testing done
- added test coverage for new component
- snapshot testing added for new component

## Screenshots
image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,8 +57,8 @@
     "test:sauce:desktop": "./script/run-nightwatch.sh --sauce -e chrome,firefox,ie,edge",
     "test:sauce:mobile": "./script/run-nightwatch.sh --sauce -e ios,android",
     "test:unit": "node ./script/run-unit-test.js",
-    "test:unit:form-system": "BABEL_ENV=test npx mocha --opts src/platform/testing/unit/mocha.opts --recursive 'src/platform/forms-system/test/js/**/*.unit.spec.js?(x)' src/platform/testing/unit/helper.js '$@'",
     "test:unit:forms": "BABEL_ENV=test npx mocha --opts src/platform/testing/unit/mocha.opts --recursive 'src/platform/forms/tests/**/*.unit.spec.js?(x)' src/platform/testing/unit/helper.js '$@'",
+    "test:unit:form-system": "BABEL_ENV=test npx mocha --opts src/platform/testing/unit/mocha.opts --recursive 'src/platform/forms-system/test/js/**/*.unit.spec.js?(x)' src/platform/testing/unit/helper.js '$@'",
```
Comment (cvalarida) [inline package.json:61]:
```suggestion
    "test:unit:forms-library": "BABEL_ENV=test npx mocha --opts src/platform/testing/unit/mocha.opts --recursive 'src/platform/forms-system/test/js/**/*.unit.spec.js?(x)' src/platform/testing/unit/helper.js '$@'",
```
I know the directory is called `forms-system`, but our word list says [forms library](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/documentation/style-guide/word-list.md#forms-library). :man_shrugging:

File: src/platform/forms/components/common/alerts/ErrorMessage.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+function ErrorMessage(props) {
+  const { active, children, message, testId, title } = props;
+
+  if (!active) return null;
+  else {
+    return (
+      <div
+        className="usa-alert usa-alert-error schemaform-failure-alert"
+        data-testid={testId}
+      >
```
Comment (cvalarida) [inline src/platform/forms/components/common/alerts/ErrorMessage.jsx:20]: The answer may be no, but does it make sense to use [Formation's `AlertBox`](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/visual-design/components/alertbox/) either here in `ErrorMessage` or in place of it?

**Review by cvalarida — APPROVED** [review]
Looks good to me. I'm wondering if we need `ErrorMessage`, though, or if we can get away with using `AlertBox` by itself. Maybe throw the `className` on it if needed? :man_shrugging:

File: src/platform/forms/components/common/alerts/ErrorMessage.jsx (hunk 2 above)
Comment (Clark-W-Griswold) [reply src/platform/forms/components/common/alerts/ErrorMessage.jsx:20]: I think that is the end goal. We have competing styles in the forms library. By starting to extract this as a reusable component, we give ourselves the opportunity to align to a source of truth and make it easier to update to that pattern.

**Review by f1337 — APPROVED** [review]
Excellent! :metal:
Bill & Ted playing air guitar, saying 'Excellent!'

**Pull request merged**